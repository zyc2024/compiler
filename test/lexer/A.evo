
int inc(int x) {
    return x+1;
}

int zero(void) { //should this syntax be legal lol
    return 0;
}

int compose(int n) {
    if (n == 0) {
        return zero();
    } else {
        return inc(compose(n-1));
    }
}

int main(char[][] args) {
    if (args[0][0] != '-') {
        return compose(0);
    } else {
        int k = 9;
        while (k < args[0][0]) {
            k = A.inc(k);
        }
        return A.inc(k);

    }
}