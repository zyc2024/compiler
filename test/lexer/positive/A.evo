
int inc(int x) {
    return x+1;
}

int zero(void) { //illegal syntax but this is lexing
    return 0;
}

int compose(int n) {
    if (n == 0) {
        return zero();
    } else {
        return inc(compose(n-1));
    }
}

int main(char[][] args) {
    if (args[0][0] != '-') {
        return compose(0);
    } else {
        int k = 9;
        while (k < args[0][0]) {
            k = A.inc(k);
        }
        return A.inc(k);

    }
}