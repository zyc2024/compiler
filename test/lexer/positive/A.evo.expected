2:1 int
2:5 id inc
2:8 (
2:9 int
2:13 id x
2:14 )
2:16 {
3:5 return
3:12 id x
3:13 +
3:14 integer 1
3:15 ;
4:1 }
6:1 int
6:5 id zero
6:9 (
6:10 void
6:14 )
6:16 {
7:5 return
7:12 integer 0
7:13 ;
8:1 }
10:1 int
10:5 id compose
10:12 (
10:13 int
10:17 id n
10:18 )
10:20 {
11:5 if
11:8 (
11:9 id n
11:11 ==
11:14 integer 0
11:15 )
11:17 {
12:9 return
12:16 id zero
12:20 (
12:21 )
12:22 ;
13:5 }
13:7 else
13:12 {
14:9 return
14:16 id inc
14:19 (
14:20 id compose
14:27 (
14:28 id n
14:29 -
14:30 integer 1
14:31 )
14:32 )
14:33 ;
15:5 }
16:1 }
18:1 int
18:5 id main
18:9 (
18:10 char
18:14 [
18:15 ]
18:16 [
18:17 ]
18:19 id args
18:23 )
18:25 {
19:5 if
19:8 (
19:9 id args
19:13 [
19:14 integer 0
19:15 ]
19:16 [
19:17 integer 0
19:18 ]
19:20 !=
19:23 character -
19:26 )
19:28 {
20:9 return
20:16 id compose
20:23 (
20:24 integer 0
20:25 )
20:26 ;
21:5 }
21:7 else
21:12 {
22:9 int
22:13 id k
22:15 =
22:17 integer 9
22:18 ;
23:9 while
23:15 (
23:16 id k
23:18 <
23:20 id args
23:24 [
23:25 integer 0
23:26 ]
23:27 [
23:28 integer 0
23:29 ]
23:30 )
23:32 {
24:13 id k
24:15 =
24:17 id A
24:18 .
24:19 id inc
24:22 (
24:23 id k
24:24 )
24:25 ;
25:9 }
26:9 return
26:16 id A
26:17 .
26:18 id inc
26:21 (
26:22 id k
26:23 )
26:24 ;
28:5 }
29:1 }
